woocommerce/woocommerce-android
# Fix RTL issues using styles, and avoid side unwanted effects over textviews across the app.

### Description
Related to task #2424 and fixes done here #4924 . Some of which had to be reverted #4930 

Long story short some of the RTL issues from #2424 were not really issues. Simply put, you can't rely on the developer setting "Force RTL layout"  to test text alignment on RTL. I really suggest reading this post before reviewing p91TBi-6aS-p2

So, most of the changes you'll see in this review are re-applying this reverted [commit](https://github.com/woocommerce/woocommerce-android/commit/a152d6d34ecff3599d6d3fd561b3acf2000b633c). 

Relevant changes are in `styles_base.xml`

### Testing instructions
1. Test that you don't see any text misaligned on regular LTR languages across the app. I did an overview of all the sections I could but I know for a fact that there are some areas like payment flows that are not that easy to setup for testing. I will ping woo-android devs to please look out during this week for any misaligned text. 
2. Then set device language to Arabic or Hebrew and check that all texts are mirrored to the right as expected. 
3. As explained in detail in the P2, keep in mind that any text that is not translated into a RTL language will not be aligned to the right as expected. For example check product list. Product name is not properly aligned until I added a product name in Hebrew: 

<img src="https://user-images.githubusercontent.com/2663464/136248858-e2f7e2ed-8650-469d-b4bf-7b4d2ac5720d.png" width="300">
